### PR TITLE
MCR 3173 fix pmd issues in mycore mods

### PR DESCRIPTION
**Pull request by Antoniafriedrich**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3173).

### CODE REVIEW

File: mycore-mods/src/main/java/org/mycore/mods/MCRMODSWrapper.java [hunk 1]
```diff
@@ -110,11 +117,14 @@ public static MCRObject wrapMODSDocument(Element modsDefinition, String projectI
      * @return true, if mods is supported
      */
     public static boolean isSupported(MCRObject obj) {
+        boolean supported;
```
Comment (rsteph-de) [inline mycore-mods/src/main/java/org/mycore/mods/MCRMODSWrapper.java]: What's the advantage of this additional variable?
We could even further simplify and remove the `if` by an OR operator
`return  isSupported(obj.getId()) || ...`

File: mycore-mods/src/main/java/org/mycore/mods/csl/MCRModsItemDataProvider.java [hunk 2]
```diff
@@ -58,46 +58,33 @@
...
     private final Set<String> nonDroppingParticles = MCRConfiguration2.getString("MCR.CSL.NonDroppingParticles")
         .stream()
         .flatMap(str -> Stream.of(str.split(",")))
         .collect(Collectors.toSet());
-
     private final Set<String> droppingParticles = MCRConfiguration2.getString("MCR.CSL.DroppingParticles")
         .stream()
         .flatMap(str -> Stream.of(str.split(",")))
         .collect(Collectors.toSet());
+    private MCRMODSWrapper wrapper;
```
Comment (rsteph-de) [inline mycore-mods/src/main/java/org/mycore/mods/csl/MCRModsItemDataProvider.java:86]: Was this code formatted? - I would expect an empty line between the variable declarations

**Review by rsteph-de — CHANGES_REQUESTED** [review]
some minor issues ...

File: mycore-mods/src/main/java/org/mycore/mods/csl/MCRModsItemDataProvider.java (hunk 2 above)
Comment (Antoniafriedrich) [reply mycore-mods/src/main/java/org/mycore/mods/csl/MCRModsItemDataProvider.java:86]: The code was formatted. I just checked again. Formatting it again does not add empty lines either.

File: mycore-mods/src/main/java/org/mycore/mods/MCRMODSLinksEventHandler.java [hunk 3]
```diff
@@ -41,9 +40,6 @@
 public class MCRMODSLinksEventHandler extends MCREventHandlerBase {
 
     public static final String INDEX_ALL_CHILDREN_PROPERTY_NAME = "MCR.MODS.LinksEventHandler.IndexAllChildren";
```
Comment (rsteph-de) [inline mycore-mods/src/main/java/org/mycore/mods/MCRMODSLinksEventHandler.java]: This constant and the value in mycore.properties are not used anymore and can be deleted

File: mycore-mods/src/main/java/org/mycore/mods/csl/MCRModsItemDataProvider.java [hunk 4]
```diff
@@ -248,76 +235,141 @@ protected void processModsPart(CSLItemDataBuilder idb) {
...
-            .map(this::getGenreStringFromElement)
-            .filter(Objects::nonNull)
-            .collect(Collectors.toSet());
-
+        final Set<String> genres = getStrings(elements);
         final List<Element> parentElements = wrapper.getElements("mods:relatedItem[@type='host']/mods:genre");
+        final Set<String> parentGenres = getStrings(parentElements);
+        String genreType = getGenreType(genres);
+
+        switch (genreType) {
```
Comment (rsteph-de) [inline mycore-mods/src/main/java/org/mycore/mods/csl/MCRModsItemDataProvider.java:243]: use switch arrow syntax:
``` java
 switch (genreType) {
            case "article" -> idb.type(handleArticleType(parentGenres));
            case "conference" -> idb.type(CSLType.PAPER_CONFERENCE);
      ...
```

**Review by rsteph-de — CHANGES_REQUESTED** [review]
2 more issues, after fixing them the PR is ready to merge

**Review by rsteph-de — APPROVED** [review]
Looks good for me now

**Pull request merged**